ml-dhs-aws/src/main/java/com/marklogic/client/dhs/CountInputStream.java

```diff
@@ -0,0 +1,37 @@
+package com.marklogic.client.dhs;
+
+import java.io.FileNotFoundException;
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class CountInputStream extends FilterInputStream {
+	public CountInputStream(InputStream inputStream) throws FileNotFoundException {
+		super(inputStream);
+		this.byteCount = 0;
+	}
+
+	private int byteCount;
+
+	public int getByteCount() {
+		return byteCount;
+	}
+
+	@Override
+	public int read() throws IOException {
+		int readLength = super.read();
+		if(readLength>0)
+			byteCount+= readLength;
+		return readLength;
+	}
+	
+	@Override
+	public int read(byte b[]) throws IOException {
+		return this.read();
+    }
+	
+	@Override
+	 public int read(byte b[], int off, int len) throws IOException {
+		return this.read();
+	}
```

Comment:
Shouldn't both of the read() overloads pass their arguments through to super.read()?
The wrapped InputStream still needs to read the CSV document in the standard way for the overload.
It might be convenient to create a private method to increment, as in
```
    private int addToByteCount(int readLength) {
        if (readLength > 0)
            byteCount += readLength;
        return readLength;
    }
```
The overrides can then be similar to:
```
    public int read(byte[] b) throws IOException {
 		return addToByteCount(super.read(b));
     }
    public int read(byte[] b, int off, int len) throws IOException {
 		return addToByteCount(super.read(b, off, len));
     }
```